spark/src/main/java/org/opensearch/sql/spark/functions/resolver/SqlTableFunctionResolver.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.functions.resolver;
+
+import static org.opensearch.sql.data.type.ExprCoreType.STRING;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.opensearch.sql.exception.SemanticCheckException;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+import org.opensearch.sql.expression.function.FunctionBuilder;
+import org.opensearch.sql.expression.function.FunctionName;
+import org.opensearch.sql.expression.function.FunctionResolver;
+import org.opensearch.sql.expression.function.FunctionSignature;
+import org.opensearch.sql.spark.client.SparkClient;
+import org.opensearch.sql.spark.functions.implementation.SqlFunctionImplementation;
+
+/**
+ * Function resolver for sql function of spark connector.
+ */
+@RequiredArgsConstructor
+public class SqlTableFunctionResolver implements FunctionResolver {
+  private final SparkClient sparkClient;
+
+  public static final String SQL = "sql";
+  public static final String QUERY = "query";
+
+  @Override
+  public Pair<FunctionSignature, FunctionBuilder> resolve(FunctionSignature unresolvedSignature) {
+    FunctionName functionName = FunctionName.of(SQL);
+    FunctionSignature functionSignature =
+        new FunctionSignature(functionName, List.of(STRING));
+    final List<String> argumentNames = List.of(QUERY);
+
+    FunctionBuilder functionBuilder = (functionProperties, arguments) -> {
```

Comment:
what argument will be supported? for example...

Reply:
For now, sql function will only take spark query argument.